### ISSUE TITLE
[mruby3] Cannot use (some of) external mgems due to symbol resolution

### ISSUE THREAD

**Issue by udzura** [mruby/mruby#5071]
Check out mruby3: // version is c1f5f9986ae4b5085bee6143d378182746462519

```
git clone https://github.com/mruby/mruby.git
cd mruby
git checkout mruby3
```

Create build_config.rb (or use `target/*`)

```
$ cat my_build_config.rb
MRuby::Build.new do |conf|
  toolchain :gcc
  conf.gem :github => 'mattn/mruby-onig-regexp'
  conf.gembox 'full-core'
  enable_debug
  #conf.enable_bintest
  #conf.enable_test
end
```

Then build,

```
$ MRUBY_CONFIG=`pwd`/my_build_config.rb rake
```

And all build is passed, but mruby would not startup due to symbol resolution in prelude.

```
$ ./bin/mruby -e 'puts "Hello World"'
undefined method 'FIND_NOT_EMPTY' for class String (NameError)
$ ./bin/mruby
undefined method 'FIND_NOT_EMPTY' for class String (NameError)
```

**Comment by matz** [comment]
Fixed by 6cd1bee574416

**Comment by udzura** [comment]
Thanks! I will investigate into new MRB_SYM() ecosystem!

**Comment by matz** [comment]
Thank **you!** And sorry for incomvienience.

**Comment by udzura** [comment]
Not OK. This is still broken:

```ruby
MRuby::Build.new do |conf|
  toolchain :gcc
  conf.gembox 'default'
  conf.gem mgem: 'mruby-simplehttpserver'
end
```

**Comment by matz** [comment]
I think it`s fixed by #5072
